Library/Homebrew/download_strategy.rb

```diff
@@ -375,35 +375,35 @@ def _fetch
       ohai "Downloading from #{url}"
     end
 
-    urls = actual_urls(url)
-    unless urls.empty?
-      ohai "Downloading from #{urls.last}"
-      if !ENV["HOMEBREW_NO_INSECURE_REDIRECT"].nil? && url.start_with?("https://") &&
-         urls.any? { |u| !u.start_with? "https://" }
-        puts "HTTPS to HTTP redirect detected & HOMEBREW_NO_INSECURE_REDIRECT is set."
-        raise CurlDownloadStrategyError, url
-      end
-      url = urls.last
-    end
-
-    curl url, "-C", downloaded_size, "-o", temporary_path
+    curl resolved_url(url), "-C", downloaded_size, "-o", temporary_path
   end
 
   # Curl options to be always passed to curl,
   # with raw head calls (`curl -I`) or with actual `fetch`.
   def _curl_opts
-    copts = []
-    copts << "--user" << meta.fetch(:user) if meta.key?(:user)
-    copts
-  end
-
-  def actual_urls(url)
-    urls = []
-    curl_args = _curl_opts << "-I" << "-L" << url
-    Utils.popen_read("curl", *curl_args).scan(/^Location: (.+)$/).map do |m|
-      urls << URI.join(urls.last || url, m.first.chomp).to_s
+    return ["--user" << meta.fetch(:user)] if meta.key?(:user)
+    []
+  end
+
+  def resolved_url(url)
+    redirect_url, _, status = Open3.capture3(*curl_args(
```

Comment:
This is an example of where I'm not sure why one of the `curl` helper methods isn't used.